cppcheckpremium-suppressions

```diff
@@ -265,3 +265,5 @@ premium-misra-cpp-2023-28.6.1
 # we use cstdio by intention
 premium-misra-cpp-2023-30.0.1
 
+# integer promotions
+premium-misra-cpp-2023-7.0.5
```

Comment:
we have ordered the suppressions in this file.. it's not random. I would like that you put this suppression between the suppressions for rules 7.0.4 and 7.0.6